From bc2b51ef9952a08260cfea5344fb72e9dbe90d57 Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Wed, 16 Apr 2025 15:11:37 -0700
Subject: [PATCH 1/5] Add MIOpenDriver dependency on SQLite::SQLite3.

* Adds "PRIVATE" to adjacent link library commands to comply with modern norms.
* Tested in TheRock with `-DMIOPEN_BUILD_DRIVER=ON` (TheRock's build environment installs no non-system development libraries, using bundled SQLite3 via cmake settings, thus flushing out issues like this which are incorrectly falling back to system libraries).

Co-Authored-By: Scott Tsai <scottt.tw@gmail.com>
---
 driver/CMakeLists.txt | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/driver/CMakeLists.txt b/driver/CMakeLists.txt
index d6ef51900..1e3d3f97f 100644
--- a/driver/CMakeLists.txt
+++ b/driver/CMakeLists.txt
@@ -72,10 +72,14 @@ if(WIN32)
 endif()
 add_dependencies(MIOpenDriver generate_kernels)
 target_include_directories(MIOpenDriver PRIVATE ../src/kernels)
-target_link_libraries(MIOpenDriver MIOpen Threads::Threads roc::rocrand)
+target_link_libraries(MIOpenDriver PRIVATE MIOpen Threads::Threads roc::rocrand nlohmann_json )
 if(NOT MIOPEN_EMBED_DB STREQUAL "")
-target_link_libraries(MIOpenDriver $<BUILD_INTERFACE:miopen_data> )
+    target_link_libraries(MIOpenDriver PRIVATE $<BUILD_INTERFACE:miopen_data> )
 endif()
+if(MIOPEN_ENABLE_SQLITE)
+    target_link_libraries(MIOpenDriver PRIVATE SQLite::SQLite3)
+endif()
+
 # Cmake does not add flags correctly for gcc
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU") 
     set_target_properties(MIOpenDriver PROPERTIES COMPILE_FLAGS -pthread LINK_FLAGS -pthread)
-- 
2.41.0.windows.1

